src/applications/MachineShopSimulator.java

```diff
@@ -138,7 +80,7 @@ void simulate(SimulationResults simulationResults) {
             Job theJob = changeState(nextToFinish);
             // move theJob to its next machine
             // decrement numJobs if theJob has finished
-            if (theJob != null && !moveToNextMachine(theJob, simulationResults))
+            if (theJob != null && !theJob.moveToNextMachine(this, simulationResults))
                 numJobs--;

```

Comment:
```suggestion
                --numJobs;
```

Feel free to ignore this – I just did for an example.